gradoop-flink/src/test/java/org/gradoop/flink/model/impl/operators/aggregation/AggregationWithPostProcessingTest.java

```diff
@@ -0,0 +1,132 @@
+   * and only once.
+   */
+  public static class SumPlusOne extends SumVertexProperty {
+
+    /**
+     * Create an instance of this test function.
+     *
+     * @param propertyKey          The property key to aggregate.
+     * @param aggregatePropertyKey The property key used to store the result.
+     */
+    public SumPlusOne(String propertyKey, String aggregatePropertyKey) {
+      super(propertyKey, aggregatePropertyKey);
+    }
+
+    /**
+     * Post-processing for this aggregate function: Increment the result by {@code 1}.
+     *
+     * @param result The result of the aggregation step.
+     * @return The final result.
+     */
+    @Override
+    public PropertyValue postAggregate(PropertyValue result) {
+      return PropertyValueUtils.Numeric.add(result, PropertyValue.create(1L));
+    }
+  }
+
+  /**
+   * Test the aggregation with a post-processing step on a logical graph.
+   *
+   * @throws Exception when the execution in Flink fails.
+   */
+  @Test
+  public void testAggregationWithPostAggregateForGraph() throws Exception {
+    FlinkAsciiGraphLoader loader = getLoaderFromString("input [" +
+      "(i1 {a: 1L}) (i2 {a: 2L}) (i3 {a: -1L}) (i4 {a: 3L})" +
+      "] expected {sum_a: 5L, sum_a_plusone: 6L} [" +
+      "(i1)(i2)(i3)(i4)" +
+      "]");
+    LogicalGraph input = loader.getLogicalGraphByVariable("input");
+    LogicalGraph expected = loader.getLogicalGraphByVariable("expected");
+    LogicalGraph result = input.aggregate(new SumVertexProperty("a", "sum_a"),
+      new SumPlusOne("a", "sum_a_plusone"));
+    collectAndAssertTrue(expected.equalsByData(result));
+  }
+
+  /**
+   * Test the aggregation with a post-processing step on a graph collection.
+   *
+   * @throws Exception when the execution in Flink fails.
+   */
+  @Test
+  public void testAggregationWithPostAggregateForGraphCollection() throws Exception {
+    FlinkAsciiGraphLoader loader = getLoaderFromString("input1 [" +
+      "(i1 {a: 1L}) (i2 {a: 2L})" +
+      "] input2 [" +
+      "(i3 {a: -1L}) (i4 {a: 3L})" +
+      "] input3 [] expected1 {sum_a: 3L, sum_a_plusone: 4L} [" +
+      "(i1)(i2)" +
+      "] expected2 {sum_a: 2L, sum_a_plusone: 3L} [" +
+      "(i3)(i4)" +
+      "] expected3 {sum_a: NULL, sum_a_plusone: NULL} []");
+    GraphCollection input = loader.getGraphCollectionByVariables("input1", "input2", "input3");
+    GraphCollection expected = loader.getGraphCollectionByVariables("expected1", "expected2",
+      "expected3");
+    GraphCollection result = input.apply(new ApplyAggregation(
+      new SumVertexProperty("a", "sum_a"),
+      new SumPlusOne("a", "sum_a_plusone")));
+    collectAndAssertTrue(expected.equalsByGraphData(result));
+  }
+
+  /**
+   * Test the aggregation with a post-processing step during graph grouping.
+   *
+   * @throws Exception when the execution in Flink fails.
+   */
+  @Test
+  public void testAggregationWithPostAggregateForGraphGrouping() throws Exception {
```

Comment:
I think this should be in `GroupingTestBase`

Reply:
This test uses the `SumPlusOne` test class, which is otherwise only used in this test class. When moving this test function, that class might have to be moved as well, I don't know if that would make sense.
Other opinions?